packages/python/plotly/plotly/express/_core.py

```diff
@@ -322,28 +327,32 @@ def make_trace_kwargs(args, trace_spec, trace_data, mapping_labels, sizeref):
                 if (
                     args["x"]
                     and args["y"]
-                    and len(trace_data[[args["x"], args["y"]]].dropna()) > 1
+                    and len(
+                        trace_data.select(nw.col(args["x"], args["y"])).drop_nulls()
+                    )
+                    > 1
                 ):
                     # sorting is bad but trace_specs with "trendline" have no other attrs
-                    sorted_trace_data = trace_data.sort_values(by=args["x"])
-                    y = sorted_trace_data[args["y"]].values
-                    x = sorted_trace_data[args["x"]].values
+                    sorted_trace_data = trace_data.sort(by=args["x"], nulls_last=True)
+                    y = sorted_trace_data.get_column(args["y"])
+                    x = sorted_trace_data.get_column(args["x"])
 
-                    if x.dtype.type == np.datetime64:
+                    if x.dtype == nw.Datetime:
                         # convert to unix epoch seconds
-                        x = x.astype(np.int64) / 10**9
-                    elif x.dtype.type == np.object_:
+                        x = x.dt.timestamp() / 10**9
```

Comment:
i'm not sure the comment is right (`# convert to unix epoch seconds`) as it assumes nanosecond resolution (which isn't necessarily the case even in pandas anymore)

maybe here we should make a helper function
```python
def _to_unix_epoch_seconds(s: nw.Series) -> nw.Series:
    dtype = s.dtype
    if dtype == nw.Date:
        return s.dt.timestamp('ms') / 1_000
    if dtype == nw.Datetime:
        if dtype.time_unit in ('s', 'ms'):
            return s.dt.timestamp('ms') / 1_000
        elif dtype.time_unit == 'us':
            return s.dt.timestamp('us') / 1_000_000
        elif dtype.time_unit == 'ns':
            return s.dt.timestamp('us') / 1_000_000_000
        else:
            msg = "Unexpected dtype, please report a bug"
            raise ValueError(msg)
    else:
        msg = f"Expected Date or Datetime, got {dtype}"
        raise TypeError(msg)
```
and then here do
```python
if x.dtype == nw.Datetime or x.dtype == nw.Date:
    x = _to_unix_epoch_seconds(x)
```